### PR TITLE
Make hasShouldComponentUpdate detect nested functions

### PR DESCRIPTION
**Pull request by adamschoenemann**
I haven't tested it thoroughly, but it works for me! :)

### CODE REVIEW

File: component.js [hunk 1]
```diff
@@ -303,10 +303,17 @@ function removeOldStaticMethods (mixins) {
 
 function hasShouldComponentUpdate (mixins) {
   return mixins.some(function (mixin) {
-    return !!mixin.shouldComponentUpdate;
+    if (mixin.shouldComponentUpdate) {
+        return true;
+    }
+    if (mixin.mixins instanceof Array) {
```
Comment (mikaelbr) [inline component.js]: Could this be changed to `Array.isArray`? instanceof checks are generally not too good in JS.

**Comment by mikaelbr** [comment]
Is this something that React supports as well? Never seen nested objects as mixins in React before. Could you possibly do a test verifying this as well?

File: component.js (hunk 1 above)
Comment (adamschoenemann) [inline component.js]: Sure, no problem

**Comment by adamschoenemann** [comment]
Well, React certainly supports nested mixins. React does not attempt to set a default `shouldComponentUpdate` via a mixin operation, so this is not really an in issue.
As a user, I would expect the omniscient `shouldComponentUpdate` would only be supplied if I had not supplied one, through nested mixins or not - but maybe that's just me!

This example works fine with React:

```javascript
var BaseHandler = {
    shouldComponentUpdate() {
        console.log('BaseHandler shouldComponentUpdate!');
        return true;
    }
};

var AppHandler = {
    mixins: [BaseHandler]
};

var FooHandler = React.createClass({
    mixins: [AppHandler],
    getInitialState() {
        return {foo: 1};
    },
    componentDidMount() {
        window.setTimeout(() => {
            this.setState({foo: this.state.foo + 1});
        }, 500);
    },
    render() {
        return DOM.div(null,
            this.state.foo,
            'FooHandler!'
        );
    }
});
```

Whereas with Omniscient, React will throw an error when attempting to set `shouldComponentUpdate`, as it is already defined by the nested mixin.

**Comment by mikaelbr** [comment]
I see. This would fix that, so very good! :cake:

**Comment by adamschoenemann** [comment]
Great! :+1:

**Pull request merged**